File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        
        int matchCount = 0;
        int zeroCount = 0;

        Set<Integer> winSet = new HashSet<>();
        for (int num : win_nums) {
            winSet.add(num);
        }

        for (int num : lottos) {
            if (num == 0) {
                zeroCount++;
            } else if (winSet.contains(num)) {
                matchCount++;
            }
        }
        
        int[] rank = {6, 6, 5, 4, 3, 2, 1};
        
        int[] answer = new int[]{rank[matchCount + zeroCount], rank[matchCount]};
        
        return answer;
    }
}

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    
    min_num = 0
    
    for i in range(6):
        if win_nums[i] in lottos:
            min_num += 1
    
    n_zero = lottos.count(0)
    
    rank = [6, 6, 5, 4, 3, 2, 1]

    return [rank[min_num + n_zero], rank[min_num]]